src/pages/settings/profile/index.jsx

```diff
@@ -0,0 +1,159 @@
+import React, { useEffect, useState } from 'react';
+import { Auth } from 'aws-amplify';
+import _ from 'lodash';
+import {
+  Form, Input, Empty, PageHeader, Card, Row, Col, Button, Space,
+} from 'antd';
+import { useRouter } from 'next/router';
+import FeedbackButton from '../../../components/FeedbackButton';
+import endUserMessages from '../../../utils/endUserMessages';
+import pushNotificationMessage from '../../../utils/pushNotificationMessage';
+
+const ProfileSettings = () => {
+  const router = useRouter();
+
+  const [user, setUser] = useState();
+  const [oldPasswordError, setOldPasswordError] = useState(null);
+  const [newPasswordError, setNewPasswordError] = useState(null);
+  const [emailError, setEmailError] = useState(null);
+
+  const initialState = {
+    changedPasswordAttributes: {},
+    changedUserAttributes: {},
+  };
+  const [newAttributes, setNewAttributes] = useState(initialState);
+  const { changedPasswordAttributes, changedUserAttributes } = newAttributes;
+
+  const setChanges = (object) => {
+    const newChanges = _.cloneDeep(newAttributes);
+    _.merge(newChanges, object);
+    setNewAttributes(newChanges);
+  };
+
+  const currentUser = () => Auth.currentAuthenticatedUser()
+    .then((userData) => setUser(userData))
+    .catch((e) => console.log('error during getuser', e));
+
+  useEffect(() => {
+    currentUser();
+  }, []);
+
+  const updateDetails = async () => {
+    const { name, email } = changedUserAttributes;
+    const { oldPassword, newPassword, confirmNewPassword } = changedPasswordAttributes;
+
+    if (name || email) {
+      setEmailError(false);
+      await Auth.updateUserAttributes(user, changedUserAttributes)
+        .then((response) => pushNotificationMessage('success', endUserMessages.DETAILS_UPDATED, 3))
+        .catch((e) => setEmailError(true));
+    }
+    if (oldPassword || newPassword || confirmNewPassword) {
+      setOldPasswordError(false);
+      setNewPasswordError(false);
+      const decimal = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[^a-zA-Z0-9])(?!.*\s).{8,}$/;
+
+      if (confirmNewPassword !== newPassword) {
+        setNewPasswordError("Passwords don't match.");
+      } else if (!newPassword.match(decimal)) {
+        setNewPasswordError('Password should include at least 8 characters, a number, special character, uppercase letter, lowercase letter.');
+      } else {
+        await Auth.changePassword(user, oldPassword, newPassword)
+          .then((response) => pushNotificationMessage('success', endUserMessages.DETAILS_UPDATED, 3))
+          .catch((error) => { setOldPasswordError("Doesn't match old password"); });
```

Comment:
Are we sure that an error always means that, or could we be misleading users?

Reply:
The 2 errors you would get are - `InvalidParameterException` when password format is invalid - less than 6 characters etc. basically wrong password as well. The other one is `InvalidPasswordException`. The rest of the errors thrown by this are here https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_ChangePassword.html. To what I see we won't see the other errors as the user is already logged in, but still should I filter the errors?

Reply:
I think that it better to show an ugly error than mislead the user. If AWS is broken and throws a 500, I think that we cannot tell the user that he mistyped the password.

Reply:
Ok, I'm printing the error message then - sometimes it could be ugly indeed
![image](https://user-images.githubusercontent.com/22479584/120787643-3b613980-c538-11eb-9dc6-39967662437b.png)
![image](https://user-images.githubusercontent.com/22479584/120787727-52079080-c538-11eb-9e61-ea8bc69d40f9.png)
Maybe it's too ugly though and should print something custom instead (invalid **username** or password when only password is typed in)